trackbase/FairRKPropagator.cxx

```diff
@@ -0,0 +1,686 @@
+    return true;
+}
+
+bool FairRKPropagator::SetDestinationPlane(TVector3& v0, TVector3& v1, TVector3& v2) {
+    fDefPlaneV0 = v0;
+    fDefPlaneV1 = v1;
+    fDefPlaneV2 = v2;
+    fPropagationFlag = TOPLANE;
+    return true;
+}
+
+bool FairRKPropagator::SetOriginPlane(TVector3& v0, TVector3& v1) {
+    LOG(warning) << "FairRKPropagator::SetLengthToPropagateTo not implemented yet";
+    return false;
+}
+
+bool FairRKPropagator::SetDestinationVolume(std::string volName, int copyNo, int option) {
+    LOG(warning) << "FairRKPropagator::SetDestinationVolume not implemented yet";
+    return false;
+}
+
+bool FairRKPropagator::SetDestinationLength(float length) {
+    LOG(warning) << "FairRKPropagator::SetDestinationLength not implemented yet";
+    return false;
+}
+
+//______________________________________________________________________________
+bool FairRKPropagator::SetPCAPropagation(int pca, int dir, FairTrackParP* par) {
+    if ( abs(dir) != 1 ) {
+        LOG(warning) << "Set PCA dir to +1 or -1.";
+        return false;
+    }
+    fPCAPropagationType = pca;
+    fPCAPropagationDir  = dir;
+    fPCAPropagationPar  = par;
+    return true;
+}
+
+//______________________________________________________________________________
+int FairRKPropagator::FindPCA(PCASetupStruct pcastruct) {
+
+    if ( pcastruct.PCA != 1 && pcastruct.PCA != 2 ) {
+        LOG(info) << "FairRKPropagator::FindPCA implemented for point (pca=1) and wire (pca=2) only";
+        return 1;
+    }
+    double charge = GetChargeFromPDG(pcastruct.PDGCode);
+    double momIn  = fPCAPropagationDir* // if set to -1, it will back propagate
+        TMath::Sqrt(fPCAPropagationPar->GetPx()*fPCAPropagationPar->GetPx()+
+                    fPCAPropagationPar->GetPy()*fPCAPropagationPar->GetPy()+
+                    fPCAPropagationPar->GetPz()*fPCAPropagationPar->GetPz());
+    if ( momIn == 0. ) return 1;
+    double vecIn[7] = {fPCAPropagationPar->GetX(),        fPCAPropagationPar->GetY(),        fPCAPropagationPar->GetZ(),
+                       fPCAPropagationPar->GetPx()/momIn, fPCAPropagationPar->GetPy()/momIn, fPCAPropagationPar->GetPz()/momIn,
+                       momIn};
+
+    double diff;
+    if ( pcastruct.PCA == 1 )
+        diff = sqrt((vecIn[0]-pcastruct.Point.X())*(vecIn[0]-pcastruct.Point.X())+
+                    (vecIn[1]-pcastruct.Point.Y())*(vecIn[1]-pcastruct.Point.Y())+
+                    (vecIn[2]-pcastruct.Point.Z())*(vecIn[2]-pcastruct.Point.Z()));
+    else //if ( pcastruct.PCA == 2 )
+        diff = CalculatePointToWireDistance(TVector3(fPCAPropagationPar->GetX(), fPCAPropagationPar->GetY(), fPCAPropagationPar->GetZ()), pcastruct.Wire1, pcastruct.Wire2, pcastruct.OnWirePCA);
+
+    fMaxStep = diff/25;
+    double res_old= diff;
+    double res = 100.0;
+    double vecOut[7];
+    double vecOutT[7];
+    for (int i=0; i< 7; i++) {vecOut[i]=0; vecOutT[i]=0;}
+
+    int nIter=0;
+    pcastruct.TrackLength = 0.;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'pcastruct.TrackLength' is assigned a value that is never used.](https://app.codacy.com/app/FairRootGroup/FairRoot/pullRequest?prid=4139720)